Parse/TPTP.cpp

```diff
@@ -2385,10 +2885,37 @@ void TPTP::term()
       _states.push(FUN_APP);
       return;
 
+    
+    case T_INTEGER_TYPE:
+    case T_REAL_TYPE:
+    case T_RATIONAL_TYPE: {
+      USER_ERROR("Polymorphic Vampire is currently not compatible with theory reasoning");
+      //the code below is in preparation for 
+      //when theorey reasoning is updated to deal with polymorphism
+      resetToks();
+      switch (tok.tag) {
+        case T_INTEGER_TYPE:
+          _termLists.push(Term::intSort());
+          break;
+        case T_REAL_TYPE:
+          _termLists.push(Term::realSort());
+          break;        
+        case T_RATIONAL_TYPE:
+          _termLists.push(Term::rationalSort());
+          break;
+        default:
+          ASSERTION_VIOLATION;
+      }
+      return;
+    }
+
     case T_STRING:
     case T_INT:
     case T_REAL:
     case T_RAT: {
+      if(env.statistics->polymorphic || env.statistics->higherOrder){
```

Comment:
I still have more than 100 files to review. Won't you have a look at this?

What's the main problem with parsing polymorphism along with theories?

(Thinking of a problem with a general theory of lists and a conjecture about a list of ints and their, say, sum.

Definitely not about making e.g. $sum polymorhic for int/rat/real.)

Reply:
The issue is that the theory reasoning has not been updated in any way to deal with truly polymorphic problems (e.g. it may make use of `NonVariableIterator`).

Until that has been dealt with, it seems safest to raise a parse error for all problems that combine polymorphism with theories (some may work, but we have no guarantees)